number_theory/modular.h

```diff
@@ -132,6 +134,19 @@ class Modular {
   }
 };
 
+// Returns the modular inverse of |number| in modulo |mod| if exists.
+// Otherwise, throws an invalid_argument exception.
+template <typename T,
+          std::enable_if_t<std::numeric_limits<T>::is_integer, bool> = true>
+T inverse(T number, T mod) {
+  T num = modular_internal::normalize(std::move(number), mod);
+  auto [x, y] = exgcd(num, mod);
+  if (x * num + y * mod != 1) {
```

Comment:
Multiplication may overflow here.

Reply:
After a second thought, I find that we don't have an overflow. For example when `T=int`, the multiplication is 32-bit. The result equals to 1 only when it is `(2**32)*k+1`, and k must be 0 in this case according to `exgcd`'s definition.
Let's note it in the comments since it is very tricky.